staging/src/k8s.io/apiserver/pkg/authentication/serviceaccount/util.go

```diff
@@ -119,19 +128,28 @@ func UserInfo(namespace, name, uid string) user.Info {
 type ServiceAccountInfo struct {
 	Name, Namespace, UID string
 	PodName, PodUID      string
+	JTI                  string
+	NodeName, NodeUID    string
```

Comment:
I think we probably should refactor this to:

```suggestion
	JTI string
	Self         Ref
	Namespace    Ref
	Pod          *Ref
	Secret       *Ref
	Node         *Ref
```

where `Ref` is

```go
type Ref struct { Name, UID string }
```

Reply:
Hm - so we don't currently have the Namespace's UID in existing call-sites/users of this structure, so I think we'd probably want to keep that as a string.

We also have previously not stored the 'Secret' information as Extra UserInfo, hence it doesn't appear here at the moment.

Do we want to change that here? If so, should that behaviour also be feature gated?

Reply:
Can we consider the refactor as something for a separate PR? Separating the changes keeps both easy to review.

Reply:
agreed, minimal diff here is helpful to review